api/client/src/intTest/java/org/projectnessie/client/auth/oauth2/ITOAuth2Client.java

```diff
@@ -162,6 +164,33 @@ void testOAuth2ClientInitialRefreshToken() {
     }
   }
 
+  @Test
+  void testOAuth2ClientIdle() {
+    OAuth2ClientParams params =
+        clientParams("Client2")
+            .idleInterval(Duration.ofSeconds(1))
+            .keepAliveInterval(Duration.ofSeconds(1))
+            .refreshSafetyWindow(Duration.ofSeconds(1))
+            .build();
+    try (OAuth2Client client = new OAuth2Client(params);
+        HttpClient validatingClient = validatingHttpClient("Client2").build()) {
+      // will fetch initial tokens and schedule refresh in 4 seconds
+      // (5 secs lifespan - 1 secs safety window)
+      client.start();
+      assertThat(client.sleeping).isFalse();
+      // after refresh tokens, will enter sleep mode (idle interval was 1 sec)
+      await().until(client.sleeping::get);
+      // thread will exit after 1 sec (keep alive interval 1 sec)
+      await().until(() -> ((OAuth2TokenRefreshExecutor) client.executor).getPoolSize() == 0);
+      // access token is now expired; calling authenticate() will wake up client, fetch new tokens
+      // immediately (on main thread), then schedule refresh, which will in turn spawn a new thread
+      AccessToken accessToken = client.authenticate();
+      assertThat(client.sleeping).isFalse();
```

Comment:
This is potentially flaky as there's no barrier that prevents the client from sleeping again and the test execution could be delayed.

Reply:
Yes, at least in theory. I am leaning in favor of deleting this test. We don't need to test the keep alive stuff, it's not our code.

Reply:
Alright, no objections so far: I'm removing this integration test.